test/integration/infrastructure/infrastructure_test.go

```diff
@@ -82,193 +80,519 @@ func validateFlags() {
+		})
+
+		It("should successfully create and delete", func() {
+			enableDnsHostnames := true
+			vpcID, err := prepareNewVPC(ctx, logger, awsClient, vpcCIDR, enableDnsHostnames)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(vpcID).NotTo(BeEmpty())
+
+			providerConfig, err := newProviderConfig(awsv1alpha1.VPC{
+				ID:               &vpcID,
+				GatewayEndpoints: []string{s3GatewayEndpoint},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			AddCleanupAction(func() {
+				err := teardownVPC(ctx, logger, awsClient, vpcID)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			namespace, err := generateNamespaceName()
+			Expect(err).NotTo(HaveOccurred())
+
+			err = runTest(ctx, logger, c, namespace, providerConfig, decoder, awsClient)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+})
+
+func runTest(ctx context.Context, logger *logrus.Entry, c client.Client, namespaceName string, providerConfig *awsv1alpha1.InfrastructureConfig, decoder runtime.Decoder, awsClient *awsclient.Client) error {
+	var (
+		namespace                 *corev1.Namespace
+		cluster                   *extensionsv1alpha1.Cluster
+		infra                     *extensionsv1alpha1.Infrastructure
+		infrastructureIdentifiers infrastructureIdentifiers
+	)
+
+	AddCleanupAction(func() {
 		By("delete infrastructure")
-		Expect(infrastructure.Delete(ctx, logger, restConfig, c, infra)).NotTo(HaveOccurred())
-		Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).NotTo(HaveOccurred())
+		Expect(client.IgnoreNotFound(c.Delete(ctx, infra))).To(Succeed())
+
+		By("wait until infrastructure is deleted")
+		err := common.WaitUntilExtensionCRDeleted(
+			ctx,
+			c,
+			logger,
+			func() extensionsv1alpha1.Object { return &extensionsv1alpha1.Infrastructure{} },
+			"Infrastructure",
+			infra.Namespace,
+			infra.Name,
+			10*time.Second,
+			16*time.Minute,
+		)
+		Expect(err).NotTo(HaveOccurred())
 
 		By("verify infrastructure deletion")
 		verifyDeletion(ctx, awsClient, infrastructureIdentifiers)
 
-		aws.InternalChartsPath = internalChartsPath
+		Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+		Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
 	})
 
-	BeforeEach(func() {
-		scheme = runtime.NewScheme()
-		Expect(awsinternal.AddToScheme(scheme)).To(Succeed())
-		Expect(awsv1alpha1.AddToScheme(scheme)).To(Succeed())
-		decoder = serializer.NewCodecFactory(scheme).UniversalDecoder()
-	})
+	By("create namespace for test execution")
+	namespace = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+	if err := c.Create(ctx, namespace); err != nil {
+		return err
+	}
 
-	Describe("#Reconcile, #Delete", func() {
-		const (
-			secretName = "cloudprovider"
+	By("create cluster")
+	cluster = &extensionsv1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+	if err := c.Create(ctx, cluster); err != nil {
+		return err
+	}
```

Comment:
```suggestion
	Expect(c.Create(ctx, cluster)).To(Succeed())
```

Reply:
etc., you can follow the same for the rest of this function and similar functions